CONTRIBUTING.md

```diff
@@ -1,51 +1,56 @@
+* Install dependencies by running `composer update` (since project does not contain `composer.lock` it's better to ensure latest versions of packages by running `update` command instead of `install`).
+* Make changes. Please remember that **all** changes have to be covered with tests.
+  * if you work on a bug fix, please start with reproducing the problem by adding failing test case(s). When you have failing test case(s), you can [create Pull Request](#opening-a-pull-request) just to reproduce fail in the CI. Then you can provide fix _in the subsequent commits_, it will make code review easier. It's allowed to modify existing test cases in bug fix Pull Request, but *only if* current behavior is proved to be invalid.
+  * if you work on existing fixers then don't change existing test cases, because these are contract between the maintainers and users (they ensure how tool works). Add new test cases that cover provided changes - preferred way of defining test cases is with [data provider](https://docs.phpunit.de/en/10.0/writing-tests-for-phpunit.html#data-providers) which uses `yield` with proper case description as a key (e.g. `yield 'Some specific scenario' => ['some', 'example', 'data'];`). Codebase may still contain test cases in different format, and it's totally acceptable to use `yield` approach next to existing `return` usages.
+* Make sure dev environment is correct: `composer dev-tools:check`.
 * Update documentation: `composer docs`. This requires the highest version of PHP supported by PHP CS Fixer. If it is not installed on your system, you can run it in a Docker container instead: `docker-compose run php-8.2 php dev-tools/doc.php`.
 * Install dev tools: `composer dev-tools:install`.
 * Run [PHPStan](https://phpstan.org/user-guide/getting-started): `composer phpstan`.
 * Check if tests pass: `composer test`.
 * Fix project itself: `composer cs:fix`.
 
-## Working With Docker
+### Opening a [Pull Request](https://help.github.com/articles/about-pull-requests/)
 
-This project provides a Docker setup that allows working on it using any of the supported PHP versions.
+You can do some things to increase the chance that your Pull Request is accepted without communication ping-pong between you and the reviewers:
 
-To use it, you first need to install:
+* Submit one Pull Request per fix or feature.
+* Don't amend commits because it makes review rounds harder - all commits from your branch will be squashed during merge, so you can treat Pull Request as a playground, without keeping everything tidy at any point.
+* If your changes are not up-to-date, [rebase](https://git-scm.com/docs/git-rebase) your branch onto the parent branch. Do it regularly whenever your branch is behind `master` branch, that will eliminate risk of problems after the merge.
+* Follow the conventions used in the project.
+* Remember about tests and documentation.
+* Don't bump version.
```

Comment:
Version of what? Can we have it specified, because I don't know what versions not to bump (dependencies in `composer.json` would be my only guess)?

Reply:
I believe it's about `PhpCsFixer\Console\Application::VERSION`.

Reply:
Let's add it then, or remove this line, having only "Don't bump version" surely will be confusing.